scenarios/apim-baseline/terraform/README.md

```diff
@@ -0,0 +1,92 @@
+   cd apim-landing-zone-accelerator/scenarios/scripts
+   ```
+
+1. Log into Azure from the AZ CLI and select your subscription.
+
+   ```bash
+   az login
+   ```
+
+1. Review and update deployment parameters.
+
+   Copy the `sample.env` into a new file called `.env` in the same directory. The main difference with the Bicep version is the need for a backend when deploying Terraform templates.
+
+   The [**.env**](../../.env) parameter file is where you can customize your deployment. The defaults are a suitable starting point, but feel free to adjust any to fit your requirements.
+
+   **Deployment parameters**
+
+    | Name  | Description | Default | Example(s) |
+    | :---- | :---------- | :------ | :--------- |
+    | `AZURE_LOCATION` | The Azure location to deploy to. | **eastus** | **westus** |
+    | `RESOURCE_NAME_PREFIX` | A suffix for naming. | **apimdemo** | **appname** |
+    | `ENVIRONMENT_TAG` | A tag that will be included in the naming. | **dev** | **stage** |
+    | `APPGATEWAY_FQDN` | The Azure location to deploy to. | **apim.example.com** | **my.org.com** |
+    | `CERT_TYPE` | selfsigned will create a self-signed certificate for the APPGATEWAY_FQDN. custom will use an existing certificate in pfx format that needs to be available in the [certs](../../certs) folder and named appgw.pfx | **selfsigned** | **custom** |
+    | `CERT_PWD` | The password for the pfx certificate. Only required if CERT_TYPE is custom. | **N/A** | **password123** |
+    | `RANDOM_IDENTIFIER` | Optional 3 character random string to ensure deployments are unique. Automatically assigned if not provided | **abc** | **pqr** |
+
+1. For terraform to work, you'll need to setup the [tf backend](https://developer.hashicorp.com/terraform/language/settings/backends/configuration). As part of the repository we provide a `azure-backend-sample.sh` script. This script will create a storage account and a container to store the terraform state. You can run the script with the following command:
+
+    ```bash
+    ./azure-backend-sample.sh \
+         --resource-group my-resource-group \
+         --storage-account my-storage-account \
+         --container my-container
+    ```
+
+1. After setting up your backend, create a `${ENVIRONMENT_TAG}-backend.hcl` file in the same directory as your `.env`. Don't include the key value, as it is hardcoded in the script. If you are using the sample script (TF Backend in Azure), the file should look like the `sample.backend.hcl` file. So if you are going to use an Azure Backend for your Terraform provider and your ENVIRONMENT_TAG is `dev`, you should have a `dev-backend.hcl` file in the same directory as your `.env` file that looks like this:
+
+   ```hcl
+   resource_group_name  = "my-resource-group"
+   storage_account_name = "mystorageaccount"
+   container_name       = "my-container"
+   ```
+
+
+1. Deploy the reference implementation.
+
+   Run the following command to deploy the APIM baseline
```

Comment:
add a note to inform users that they will encounter prompts during the script execution and need to respond with `y` to proceed.